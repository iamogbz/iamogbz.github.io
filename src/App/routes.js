import homeRoute from "../Home/routes";
import labRoutes from "../Labs/routes";

const route = homeRoute.addNestedRoutes(labRoutes);

export default route;
