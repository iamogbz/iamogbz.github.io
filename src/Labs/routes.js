import Chart from "react-router-chart";

import Labs from ".";

export default Chart.route({ name: "labs" })
    .rPath("labs")
    .rComponent(Labs)
    .rExact(true);
