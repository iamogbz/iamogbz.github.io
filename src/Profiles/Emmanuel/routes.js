import Chart from "react-router-chart";

import Profile from ".";

export default Chart.route({ name: "emmanuel" })
    .rPath(".emmanuel")
    .rComponent(Profile)
    .rExact(true);
