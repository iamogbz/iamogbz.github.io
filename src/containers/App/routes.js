import homeRoute from "../Home/routes";
import labRoutes from "../Labs/routes";
import profileRoutes from "../Profiles/routes";

const route = homeRoute.addNestedRoutes(labRoutes, profileRoutes);

export default route;
