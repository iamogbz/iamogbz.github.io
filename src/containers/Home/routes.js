import Chart from "react-router-chart";

import Home from "./Loadable";

export default Chart.route({ name: "home" })
    .rPath("/")
    .rComponent(Home)
    .rExact(true);
