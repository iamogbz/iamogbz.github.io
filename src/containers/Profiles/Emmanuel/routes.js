import Chart from "react-router-chart";

import Profile from "./Loadable";

export default Chart.route({ name: "emmanuel" })
    .rPath(".emmanuel")
    .rComponent(Profile)
    .rExact(true);
