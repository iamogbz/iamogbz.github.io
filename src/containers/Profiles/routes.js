import emmanuelRoutes from "./Emmanuel/routes";

const defaultRoute = {
    path: "profile",
    children: [...emmanuelRoutes],
};

export default [
    defaultRoute,
    // Support legacy links e.g. #/profile.emmanuel
    ...defaultRoute.children.map(route => ({
        ...route,
        path: `${defaultRoute.path}.${route.path}`,
    })),
];
