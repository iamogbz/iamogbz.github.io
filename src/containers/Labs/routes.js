import Chart from "react-router-chart";

import Labs from "./Loadable";

export default Chart.route({ name: "labs" })
    .rPath("labs")
    .rComponent(Labs)
    .rExact(true);
