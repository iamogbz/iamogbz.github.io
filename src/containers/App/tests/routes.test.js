import routes from "../routes";

describe("Routes", () => {
    it("matches render snapshot", () => {
        expect(routes).toMatchSnapshot();
    });
});
