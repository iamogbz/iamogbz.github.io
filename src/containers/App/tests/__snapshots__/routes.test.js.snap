// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes matches render snapshot 1`] = `
Array [
  <t
    component={[Function]}
    exact={true}
    path="/"
  />,
  <t
    component={[Function]}
    exact={true}
    path="/labs"
  />,
  <t
    component={[Function]}
    exact={true}
    path="/profile.emmanuel"
  />,
]
`;
